### PR TITLE
ESBJAVA-4409

### PR DESCRIPTION
**Pull request by erandacr**

### CODE REVIEW

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java [hunk 1]
```diff
@@ -58,6 +63,10 @@ public JMSPollingConsumer( Properties jmsProperties, long scanInterval, String n
         strUserName = jmsProperties.getProperty(JMSConstants.PARAM_JMS_USERNAME);
         strPassword = jmsProperties.getProperty(JMSConstants.PARAM_JMS_PASSWORD);
         this.name = name;
+        this.retryIteration = 0;
+        this.reconnectionProgressionFactor = 2.0;
+        this.maxReconnectDuration = 60000;
```
Comment (nadeeshaan) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java]: Better if we can use constants in this case

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java [hunk 2]
```diff
@@ -228,6 +255,27 @@ public Message poll() {
...
+                        logger.info("InitialReconnectDuration reached to MaxReconnectDuration.");
+                    }
+                    logger.error("Reconnection attempt : " + (retryIteration++) + " for JMS Inbound : " +
+                            name + " failed. Next retry in " + (retryDuration / 1000) +
+                            " seconds");
+                }
+                try {
+                    Thread.sleep(retryDuration);
+                } catch (InterruptedException ignore) {
+                }
```
Comment (nadeeshaan) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java]: Better if we can add a comment here mentioning, why we are specifically ignoring this exception

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java [hunk 3]
```diff
@@ -68,6 +77,16 @@ public JMSPollingConsumer( Properties jmsProperties, long scanInterval, String n
                 iReceiveTimeout = null;
             }
         }
+
+        String strReconnectDuration = jmsProperties.getProperty(JMSConstants.JMS_RETRY_DURATION);
+        if (strReconnectDuration != null) {
+            try {
+                this.reconnectDuration = Long.parseLong(strReconnectDuration.trim());
+            } catch (NumberFormatException e) {
+                logger.warn("Invalid value for transport.jms.retry.duration : " + strReconnectDuration);
```
Comment (nuwanw) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java]: better adding the exception message as well

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java (hunk 1 above)
Comment (erandacr) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java]: Fixed with https://github.com/wso2/carbon-mediation/pull/748

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java (hunk 2 above)
Comment (erandacr) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSPollingConsumer.java]: Fixed with https://github.com/wso2/carbon-mediation/pull/748

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSConstants.java [hunk 4]
```diff
@@ -179,7 +179,9 @@
 
     public static final String HYPHEN_REPLACEMENT_STR = "_DASHED_";
 
-    public static final String DEFAULT_HYPHEN_SUPPORT = HYPHEN_MODE_NONE;     
+    public static final String DEFAULT_HYPHEN_SUPPORT = HYPHEN_MODE_NONE;
+
+    public static final String JMS_RETRY_DURATION = "transport.jms.retry.duration";
```
Comment (Niranjan-K) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSConstants.java]: We do not introduce new APIs in patch fixes. Not sure if this for a patch fix, but if this minor then it is fine.

**Pull request merged**